### PR TITLE
Docker cleanup and multiple platforms

### PR DESCRIPTION
**Pull request by treyrich**
This updates the docker workflow to be simpler and more compliant to standard-practices. The goals of this PR are:
1. Reduce to a single Dockerfile
2. Remove pm2 from Docker and reduce to a single-process image
3. Remove need for Typescript building on every startup of a container
4. Support multi-platform builds
5. Reduce size of docker image

### CODE REVIEW

File: .github/workflows/docker-publish-njsPC-linux.yml [hunk 1]
```diff
@@ -11,71 +11,22 @@ jobs:
   build-and-push:
     runs-on: ubuntu-latest
```
Comment (tagyoureit) [inline .github/workflows/docker-publish-njsPC-linux.yml:12]: Is this still valid with the 3 linux builds?

File: .github/workflows/docker-publish-njsPC-linux.yml [hunk 2]
```diff
@@ -11,71 +11,22 @@ jobs:
...
       - name: Build and push combined Docker image
-        uses: docker/build-push-action@v4
+        uses: docker/build-push-action@v6
         with:
-          context: .
-          file: ./.docker/Dockerfile.linux
           push: true
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
           tags: |
             tagyoureit/njspc-dp-combined-app:latest
```
Comment (tagyoureit) [inline .github/workflows/docker-publish-njsPC-linux.yml:32]: Please rename to remove the combined tag.

File: .github/workflows/docker-publish-njsPC-linux.yml (hunk 1 above)
Comment (treyrich) [reply .github/workflows/docker-publish-njsPC-linux.yml:13]: Yes, this is just the platform that the runner uses, not the resulting images.

File: .github/workflows/docker-publish-njsPC-linux.yml (hunk 2 above)
Comment (treyrich) [reply .github/workflows/docker-publish-njsPC-linux.yml]: I've renamed to just njspc, please let me know if you'd like a different name. You'll need to setup the repository on dockerhub before we merge this.

**Comment by treyrich** [comment]
I have added some tagging logic to tag things on the docker repo. I just took the default settings from docker's example which I think is a reasonable starting place, but it can be refined. By default it will build an image for each push on the master branch (tagged as `master` and the sha hash), and on new version tags (tagged as the version and `latest).

Overall assuming you're ok with the repository name, I think that this is ready to go. I've run some tests [on my own repo](https://hub.docker.com/repository/docker/treyrich2001/njspc) and everything appears to be working as expected.

**Comment by treyrich** [comment]
I've created a [PR](https://github.com/rstrouse/nodejs-poolController-dashPanel/pull/86) over on the dash panel repo. Not sure if we were waiting to merge this one until that was ready?

**Comment by tagyoureit** [comment]
No, it's just me!  I created the Docker repo 'njspc'.

**Comment by tagyoureit** [comment]
Just published v8.0.4 https://github.com/tagyoureit/nodejs-poolController/releases/tag/v8.0.4 to test this out.

**Comment by treyrich** [comment]
Perfect, looks like it’s working properly!

**Comment by tagyoureit** [comment]
Is it supposed to create all these tags?  
- latest
- sha-xxxxxxxx
- 8
- 8.0
- 8.0.4
- master

image

image

**Comment by treyrich** [comment]
Yes, the logic is:
- master: someone can track the master branch changes
- sha: someone can target a specific commit
- 8: anything in 8.X.X
- 8.0: anything in 8.0.X
- 8.0.4: target a specific release
- latest: always have the latest tagged release

it feels like a lot, but they’re actually all really useful for targeting different things.

**Pull request merged**